modules/Bio/EnsEMBL/DBSQL/DBConnection.pm

```diff
@@ -713,6 +713,47 @@ sub prepare {
    return $sth;
 }
 
+
+=head2 prepare_cached
+
+  Arg [1]    : string $string
+               the SQL statement to prepare
+  Example    : $sth = $db_connection->prepare_cached("SELECT column FROM table");
+  Description: Prepares a SQL statement using the internal DBI database handle
+               and returns the DBI statement handle. The prepared statement is
+               cached so that it does not have to be prepared again.
+  Returntype : DBI statement handle
+  Exceptions : thrown if the SQL statement is empty, or if the internal
+               database handle is not present
+  Caller     : Adaptor modules
+  Status     : Experimental
+
+=cut
+
+sub prepare_cached {
+   my ($self,@args) = @_;
+
+   if( ! $args[0] ) {
+     throw("Attempting to prepare_cached an empty SQL query.");
+   }
+
+   #warn "SQL(".$self->dbname."):" . join(' ', @args) . "\n";
+   if ( ($self->reconnect_when_lost()) and (!$self->db_handle()->ping()) ) {
```

Comment:
1. Parts of this might be redundant - as per our conversation a moment ago, calling DBConnection::db_handle() already involves a connection attempt.
2. Have you checked if this actually does anything when _disconnect_when_inactive()_ is enabled?
3. Please check the  attribute _Active_ of the database handle rather than the return value of _ping()_ to check if we are connected to the database, the latter is not guaranteed to give the correct answer - especially for MySQL.